cluster_experiments/experiment_analysis.py

```diff
@@ -111,6 +114,25 @@ def get_point_estimate(self, df: pd.DataFrame) -> float:
         self._data_checks(df=df)
         return self.analysis_point_estimate(df)
 
+    def pvalue_based_on_hypothesis(self, model_result) -> float:
+        """Returns the p-value of the analysis
+        Arguments:
+            model_result: statsmodels result object
+            verbose (Optional): bool, prints the regression summary if True
+
+        """
+        treatment_effect = model_result.params[self.treatment_col]
+        p_value_half = model_result.pvalues[self.treatment_col] / 2
+
+        if self.hypothesis == "less":
+            p_value = p_value_half if treatment_effect <= 0 else 1 - p_value_half
```

Comment:
instead of computing p_value and returning in the end, I think you cna can just return here